src/agent/sqlite_manager/src/sqlite_manager.cpp

```diff
@@ -0,0 +1,210 @@
+#include <sqlite_manager.hpp>
+
+#include <format>
+#include <iostream>
+
+namespace sqlite_manager
+{
+    SQLiteManager::SQLiteManager(const std::string& dbName)
+        : m_dbName(dbName)
+        , m_db(make_unique<SQLite::Database>(dbName, SQLite::OPEN_READWRITE | SQLite::OPEN_CREATE))
+    {
+        // Open the database in WAL mode
+        m_db->exec("PRAGMA journal_mode=WAL;");
+    }
+
+    int SQLiteManager::CreateTable(const std::string& tableName, const std::vector<Col>& cols)
+    {
+        std::string queryString = "CREATE TABLE IF NOT EXISTS " + tableName + " (";
+
+        for (const Col& col : cols)
+        {
+            queryString += col.m_name + " " + mapColTypeString.at(col.m_type);
+            if (col.m_primaryKey)
+                queryString += " PRIMARY KEY";
+            if (col.m_autoIncrement)
+                queryString += " AUTOINCREMENT";
+            if (col.m_notNull)
+                queryString += " NOT NULL";
+            queryString += ", ";
+        }
+        // Remove the trailing ", "
+        queryString.erase(queryString.size() - 2);
+        queryString += ");";
+
+        std::cout << "QueryString: " + queryString << std::endl;
+
+        std::lock_guard<std::mutex> lock(m_mutex);
+        try
+        {
+            m_db->exec(queryString);
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Error initializing table: " << e.what() << std::endl;
+            throw;
+        }
+        return 1;
+    }
+
+    int SQLiteManager::Insert(const std::string& tableName, const std::vector<Col>& cols)
```

Comment:
Same. If we launch an exception in case of error, I think a return value is not necessary.

Reply:
Done